### PR TITLE
chore: update rollup build

### PR DESCRIPTION
**Pull request by carlbrugger**
## Tell code reviewer what to test:
Running the recordHook with autocast and/or foreign-constraints should not produce `TypeError: Cannot read properties of undefined (reading 'get')` or `event.fetch error:  Error: HTTP error! Status: 400`.

## Summary:
This release fixes the builds of plugins utilizing the recordHook package by including recordHook as an external. This also moves the listener dep to peerDependencies which are considered externals.

Closes https://github.com/FlatFilers/support-triage/issues/1067
Closes https://github.com/FlatFilers/support-triage/issues/1045

### CODE REVIEW

File: plugins/record-hook/package.json [hunk 1]
```diff
@@ -46,20 +46,13 @@
...
   "devDependencies": {
-    "@flatfile/utils-testing": "^0.1.3",
-    "@rollup/plugin-commonjs": "^25.0.7",
-    "@rollup/plugin-json": "^6.0.1",
-    "@rollup/plugin-node-resolve": "^15.2.3",
-    "@rollup/plugin-terser": "^0.4.4",
-    "@rollup/plugin-typescript": "^11.1.5",
-    "rollup": "^4.3.0",
-    "rollup-plugin-dts": "^6.1.0"
+    "@flatfile/api": "^1.7.4"
```
Comment (carlbrugger) [inline plugins/record-hook/package.json:56]: The recordHook only `import type ... from '@flatfile/api'`

File: plugins/autocast/package.json [hunk 2]
```diff
@@ -42,15 +42,10 @@
...
     "@flatfile/util-common": "^1.0.0"
   },
-  "devDependencies": {
-    "@rollup/plugin-commonjs": "^25.0.7",
-    "@rollup/plugin-json": "^6.0.1",
-    "@rollup/plugin-node-resolve": "^15.2.3",
-    "@rollup/plugin-typescript": "^11.1.5",
-    "rollup": "^4.3.0"
+  "peerDependencies": {
+    "@flatfile/plugin-record-hook": "^1.4.4",
```
Comment (carlbrugger) [inline plugins/autocast/package.json:48]: Without the `@flatfile/plugin-record-hook` as a `peerDependencies`, two versions of the recordHook are generated (`const BulkRecordHook = ...` and `const dist_BulkRecordHook = ...`) causing the `event.fetch error:  Error: HTTP error! Status: 400` error.

File: plugins/record-hook/rollup.config.mjs [hunk 3]
```diff
@@ -23,15 +23,12 @@ const external = [
 
 function commonPlugins(browser, umd = false) {
   return [
-    !umd
-      ? peerDepsExternal({
-          includeDependencies: true,
-        })
-      : undefined,
+    !umd ? peerDepsExternal() : undefined,
```
Comment (carlbrugger) [inline plugins/record-hook/rollup.config.mjs:26]: `includeDependencies: true,` is deprecated and using it results in `const sheet = api.sheets.get(...)` to return `TypeError: Cannot read properties of undefined (reading 'get')`

File: plugins/record-hook/package.json [hunk 4]
```diff
@@ -46,20 +46,11 @@
...
-    "@flatfile/utils-testing": "^0.1.3",
-    "@rollup/plugin-commonjs": "^25.0.7",
-    "@rollup/plugin-json": "^6.0.1",
-    "@rollup/plugin-node-resolve": "^15.2.3",
-    "@rollup/plugin-terser": "^0.4.4",
-    "@rollup/plugin-typescript": "^11.1.5",
-    "rollup": "^4.3.0",
-    "rollup-plugin-dts": "^6.1.0"
+  "peerDependencies": {
+    "@flatfile/api": "^1.7.4",
```
Comment (carlbrugger) [inline plugins/record-hook/package.json:53]: recordHook only imports types from `@flatfile/api`

**Comment by bangarang** [comment]
@SocketSecurity ignore npm/esbuild@0.17.19

**Review by bangarang — APPROVED** [review]

**Review by bangarang — APPROVED** [review]

**Pull request merged**